adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -1094,25 +1074,49 @@ protected final Future<CommandConsumer> createCommandConsumer(
                     waitForCommandSpan.finish();
                 })
                 .map(consumer -> {
-                    if (!requestProcessed.get()) {
-                        // if the request was not responded already, add a timer for triggering an empty response
-                        addCommandReceptionTimer(ctx, requestProcessed, responseReady, ttdSecs, waitForCommandSpan);
-                    }
                     // wrap the consumer to be able to create a separate span on closing the consumer (errors should just be logged on that span, not the parent)
-                    return new CommandConsumer() {
+                    final ResponseReadyTracker responseReadyTracker = new ResponseReadyTracker() {
+
+                        private final Future<Void> command = responseReady.future()
+                            .compose(ar -> closeCommandConsumer())
+                            .recover(thr -> closeCommandConsumer().compose(v -> Future.failedFuture(thr)));
```

Comment:
FMPOV, the variable name doesn't fit well here and it would be better to inline the variable value in the `waitForCommand` method.

Reply:
I still think the `command` or now `waitForCommandFuture` field isn't needed here and its value could be inlined.

In any case, I think it would be good to restore this comment from the old code, putting it before the closeCommandConsumer invocation:
```
// command was received or ttd has timed out
// we wait for the CommandConsumer having been closed before delivering the response to the
// device in order to prevent a race condition when the device immediately sends a new
// request and the CommandConsumer from the current request has not been closed yet
```

Reply:
I did it like that in order to compose `closeCommandConsumer` when the `ResponseReadyTracker` is created.
This way even if caller of `createResponseReadyTracker` "forget" to call `waitFor`  the consumer would be closed when `responseReady` is completed. 
At the moment seems that `doUpdateMessage` always call `waitFor` but in future modification it might not be the case. So I consider this implementation as more safe.

Reply:
Ok, good.